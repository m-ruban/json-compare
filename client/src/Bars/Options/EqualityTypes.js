import React from 'react'
import {FormGroup, Badge, Input, Form, UncontrolledTooltip} from 'reactstrap'
import { AppConsumer } from '../../Contexts/Provider'

const EqualityTypes = ({isCompare, toggleCompare}) => (
  <AppConsumer>
    {
      ({ equalityTypes, changeEqualityTypes }) => (
        <Form className="settings" inline>
          <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
            <Input
              type="select"
              id="typeCompare"
              bsSize="sm"
              value={equalityTypes}
              onChange={changeEqualityTypes}
            >
              <option value='=='>==</option>
              <option value='==='>===</option>
              <option value='ignore'>Ignore</option>
            </Input>
            <Badge id="tooltipTypeCompare" color="success" pill>?</Badge>
            <UncontrolledTooltip target="tooltipTypeCompare">
              == - not strict equality<br />
              === - strict equality<br />
              Ignore - comparison ignores values (only keys are compared)
            </UncontrolledTooltip>
          </FormGroup>
        </Form>
      )
    }
  </AppConsumer>
)

export default EqualityTypes
