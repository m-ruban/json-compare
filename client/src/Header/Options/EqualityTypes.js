import React from 'react';
import { FormGroup, Badge, Input, Form, UncontrolledTooltip } from 'reactstrap';
import { AppConsumer } from '../../Contexts/Provider';

const EqualityTypes = () => (
  <Form className="settings" inline>
    <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
      <AppConsumer>
        {({ equalityTypes, changeEqualityTypes }) => (
          <Input
            type="select"
            id="typeCompare"
            bsSize="sm"
            value={equalityTypes}
            onChange={changeEqualityTypes}
          >
            <option value="==">==</option>
            <option value="===">===</option>
            <option value="ignore">Ignore</option>
          </Input>
        )}
      </AppConsumer>
      <Badge id="tooltipTypeCompare" color="success" pill>
        ?
      </Badge>
      <UncontrolledTooltip target="tooltipTypeCompare">
        == - not strict equality
        <br />
        === - strict equality
        <br />
        Ignore - comparison ignores values (only keys are compared)
      </UncontrolledTooltip>
    </FormGroup>
  </Form>
);

export default EqualityTypes;
