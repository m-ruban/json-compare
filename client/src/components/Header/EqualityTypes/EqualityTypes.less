@import "~styles/variables.less";

.settings {
  margin-top: 0;

  &__types {
    border-color: @border-color !important;
    background-color: @secondary-background-color !important;
    color: @primary-color !important;
    padding: 0 0.5rem;
    height: calc(1.6rem + 2px);
    margin-left: 0.5rem;
  }
}
