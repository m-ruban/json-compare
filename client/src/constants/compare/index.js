export const COMPARE_DIFF = 'diff';
export const COMPARE_REQ = 'req';
