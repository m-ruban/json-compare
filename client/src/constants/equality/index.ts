export const EQUALITY_TYPE_STRICT = 'STRICT';
export const EQUALITY_TYPE_NOT_STRICT = 'NOT_STRICT';
export const EQUALITY_TYPE_IGNORE = 'IGNORE';
