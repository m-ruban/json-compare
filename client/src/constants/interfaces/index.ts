export interface IAlert {
    res: string;
    scalar: boolean;
}

export interface IAlertMap {
    alerts: Map<string, IAlert>;
}

export interface ILog {
    text: string;
    type: string;
}
