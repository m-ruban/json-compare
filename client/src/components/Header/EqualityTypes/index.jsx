import React from 'react';
import PropTypes from 'prop-types';
import { FormGroup, Input, Form } from 'reactstrap';

import withConsumer from 'hoc/withConsumer';
import { EQUALITY_TYPE_STRICT, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_IGNORE } from 'constants/equality';
import 'components/Header/EqualityTypes/EqualityTypes.less';

const EqualityTypes = ({ equalityTypes, changeEqualityTypes }) => (
    <Form className="settings" inline>
        <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
            <Input
                type="select"
                className="settings__types"
                bsSize="sm"
                value={equalityTypes}
                onChange={changeEqualityTypes}
            >
                <option value={EQUALITY_TYPE_STRICT}>{EQUALITY_TYPE_STRICT}</option>
                <option value={EQUALITY_TYPE_NOT_STRICT}>{EQUALITY_TYPE_NOT_STRICT}</option>
                <option value={EQUALITY_TYPE_IGNORE}>{EQUALITY_TYPE_IGNORE}</option>
            </Input>
        </FormGroup>
    </Form>
);

EqualityTypes.propTypes = {
    equalityTypes: PropTypes.string,
    changeEqualityTypes: PropTypes.func,
};

export default withConsumer(EqualityTypes);
