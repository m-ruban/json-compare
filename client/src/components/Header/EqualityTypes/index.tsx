import * as React from 'react';
import { Form, FormGroup, Input } from 'reactstrap';

import 'components/Header/EqualityTypes/EqualityTypes.less';
import { EQUALITY_TYPE_IGNORE, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_STRICT } from 'constants/equality';
import withConsumer from 'hoc/withConsumer';

interface IEqualityTypesProps {
    equalityTypes: string;
    changeEqualityTypes: (event: React.SyntheticEvent) => void;
}

const EqualityTypes = ({ equalityTypes, changeEqualityTypes }: IEqualityTypesProps) => (
    <Form className="settings" inline>
        <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
            <Input
                type="select"
                className="settings__types"
                bsSize="sm"
                value={equalityTypes}
                onChange={changeEqualityTypes}
            >
                <option value={EQUALITY_TYPE_STRICT}>{EQUALITY_TYPE_STRICT}</option>
                <option value={EQUALITY_TYPE_NOT_STRICT}>{EQUALITY_TYPE_NOT_STRICT}</option>
                <option value={EQUALITY_TYPE_IGNORE}>{EQUALITY_TYPE_IGNORE}</option>
            </Input>
        </FormGroup>
    </Form>
);

export default withConsumer(EqualityTypes);
