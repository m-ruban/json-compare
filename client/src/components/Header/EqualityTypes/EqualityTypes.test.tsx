import React from 'react';

import EqualityTypes from 'components/Header/EqualityTypes';
import { EQUALITY_TYPE_IGNORE } from 'constants/equality';
import { render } from 'enzyme';
import { Provider } from 'store/Context';

export const equalityTypesIgnore = {
    equalityTypes: EQUALITY_TYPE_IGNORE,
};

describe('<EqualityTypes />', () => {
    it('Renders correctly', () => {
        const output = render(
            <Provider value={equalityTypesIgnore}>
                <EqualityTypes />
            </Provider>,
        );
        expect(output).toMatchSnapshot();
    });
});
