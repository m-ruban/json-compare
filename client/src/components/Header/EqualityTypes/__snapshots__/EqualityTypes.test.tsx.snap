// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EqualityTypes /> Renders correctly 1`] = `
<form
  class="settings form-inline"
>
  <div
    class="mb-2 mr-sm-2 mb-sm-0 form-group"
  >
    <select
      class="settings__types form-control-sm form-control"
    >
      <option
        value="STRICT"
      >
        STRICT
      </option>
      <option
        value="NOT_STRICT"
      >
        NOT_STRICT
      </option>
      <option
        selected=""
        value="IGNORE"
      >
        IGNORE
      </option>
    </select>
  </div>
</form>
`;
